packages/typespec-ts/src/index.ts

```diff
@@ -304,6 +304,9 @@ export async function $onEmit(context: EmitContext) {
     }
 
     binder.resolveAllReferences(modularSourcesRoot);
+    if (program.compilerOptions.noEmit || program.hasError()) {
```

Comment:
`program.hasError()` is for what? does it mean compiler error or sth? i just notice if there is error thrown we would fail the generation directly.